rewrite-java/src/main/java/org/openrewrite/java/JavaTemplate.java

```diff
@@ -247,13 +246,85 @@ public J visitMethodDeclaration(J.MethodDeclaration method, Integer p) {
                             return method.withBody(autoFormat(body, p, getCursor()));
                         }
                         case METHOD_DECLARATION_PARAMETERS: {
-                            return method.withParameters(substitutions.unsubstitute(templateParser.parseParameters(substitutedTemplate)));
+                            List<Statement> parameters = substitutions.unsubstitute(templateParser.parseParameters(substitutedTemplate));
+
+                            // Update the J.MethodDeclaration's type information to reflect its new parameter list
+                            JavaType.Method newType = method.getType();
+                            if(newType != null) {
+                                List<String> paramNames = new ArrayList<>();
+                                List<JavaType> paramTypes = new ArrayList<>();
+                                for(Statement parameter : parameters) {
+                                    if(!(parameter instanceof J.VariableDeclarations)) {
```

Comment:
I can't remember the scenario, but this is likely not true, or I think we would have made MethodDeclaration parameters a list of `J.VariableDeclarations`

Reply:
Looking at the language spec for Java 11 it looks to me like J.VariableDeclarations is the only option, here: https://docs.oracle.com/javase/specs/jls/se11/html/jls-8.html#jls-8.4.1